homeassistant/components/tts/__init__.py

```diff
@@ -387,7 +391,16 @@ class Provider(object):
     """Represent a single provider."""
 
     hass = None
-    language = None
+
+    @property
+    def language(self):
+        """Default language."""
+        return None
+
+    @property
+    def supported_languages(self):
+        """List of supported languages."""
+        return None
```

Comment:
You are not wrong with it. I do that while the default handling on all base entities do that in this way. HomeAssistant set property to default None and function to raise a not implement exception.